homeassistant/components/influxdb/sensor.py

```diff
@@ -104,32 +180,54 @@ def __init__(self, hass, influx_conf, query):
             self._value_template.hass = hass
         else:
             self._value_template = None
-        database = query.get(CONF_DB_NAME)
         self._state = None
         self._hass = hass
 
-        where_clause = query.get(CONF_WHERE)
-        where_clause.hass = hass
-
-        influx = InfluxDBClient(
-            host=influx_conf["host"],
-            port=influx_conf["port"],
-            username=influx_conf["username"],
-            password=influx_conf["password"],
-            database=database,
-            ssl=influx_conf["ssl"],
-            verify_ssl=influx_conf["verify_ssl"],
-        )
+        if use_v2_api:
+            influx = InfluxDBClientV2(**influx_conf)
```

Comment:
I think we should try to reuse the client from the integration init module. The config in the init module should configure the client, while the sensor platform config just configures the queries.

Reply:
I thought about this but that would be a breaking change. I think its pretty safe to assume that the most common use case here is to have both the sensor and the recorder component pointing to the same InfluxDB deployment.  

However that isn't a requirement today. Today I can create sensors just to report data in an InfluxDB deployment that HA doesn't write to at all.  I'm not required to only read from the same InfluxDB deployment that HA is writing to. I can have multiple sets of sensors each pointing to a different InfluxDB deployment.  I don't even necessarily need to give HA write access to my InfluxDB deployment if I just want sensors.

This idea would certainly simplify the code but I wasn't comfortable making this kind of breaking change as my own decision.  If you all think its the right call I can do it but want to make sure everyone is on the same page with what it would mean for users.

Reply:
The best thing would be if we could configure multiple clients and select which ones to give write access to, between 0 and all.

But yeah, it's a too big change for this PR, I think. Let's scrap it for now.